### PR TITLE
chore(ci): test with lua-resty-events 0.2.1 and OpenResty 1.25.3.1

### PR DESCRIPTION
**Pull request by chronolaw**
lua-resty-events has upgraded to 0.2.1, we should test this library with latest version.

https://github.com/Kong/lua-resty-events/releases/tag/0.2.1

KAG-4605

### CODE REVIEW

File: .github/workflows/build_and_test_with_resty_events.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        openresty-version: [1.19.9.1, 1.21.4.1]
+        openresty-version: [1.19.9.1, 1.21.4.1, 1.25.3.1]
```
Comment (bungle) [inline .github/workflows/build_and_test_with_resty_events.yml:23]: I would include `1.21.4.3` too here, or replace `1.21.4.1` with it.

**Review by bungle — APPROVED** [review]
Other than the comment below, approved.

**Review by bungle — APPROVED** [review]

**Review by outsinre — APPROVED** [review]

**Comment by chronolaw** [comment]
Some ideas: perhaps we should add the versions of  lua-resty-events library to test matrix, not only for one tag.

**Comment by windmgc** [comment]
@chronolaw Just FYI we're using lua-resty-healthcheck 1.6.x for gateway 3.4/3.5 and 3.0.x for gateway >=3.6. So we might want to test that against `release/1.6.x` as well

**Comment by chronolaw** [comment]
> @chronolaw Just FYI we're using lua-resty-healthcheck 1.6.x for gateway 3.4/3.5 and 3.0.x for gateway >=3.6. So we might want to test that against `release/1.6.x` as well

If this PR is merged I will back port it.

**Review by locao — APPROVED** [review]
We can do it later as this workflow doesn't run frequently, but we're probably good to stop testing with OpenResty 1.19.9.1.

**Pull request merged**